### PR TITLE
Replace `Cfg_with_liveness` with `Cfg_with_infos`

### PR DESCRIPTION
**Pull request by xclerc**
This pull request extends `Cfg_with_liveness` to
not only hold liveness (besides the CFG and its
layout), but also dominators and loop infos.

As a consequence, the module is also renamed,
to `Cfg_with_infos`.

### CODE REVIEW

File: backend/regalloc/regalloc_irc.ml [hunk 1]
```diff
@@ -508,13 +511,12 @@ let run : Cfg_with_infos.t -> Cfg_with_infos.t =
...
   | [] -> ()
-  | _ :: _ as spilled_nodes -> (
+  | _ :: _ as spilled_nodes ->
     List.iter spilled_nodes ~f:(fun reg -> State.add_spilled_nodes state reg);
     (* note: rewrite will remove the `spilling` registers from the "spilled"
        work list and set the field to unknown. *)
-    match rewrite state cfg_with_infos ~spilled_nodes ~reset:false with
-    | false -> ()
-    | true -> Cfg_with_infos.invalidate_liveness cfg_with_infos));
+    let (_ : bool) = rewrite state cfg_with_infos ~spilled_nodes ~reset:false in
```
Comment (gretay-js) [inline backend/regalloc/regalloc_irc.ml:518]: The result of `rewrite` is always `false` here I think. Maybe add an `assert` instead of ignoring it?

Comment (xclerc) [reply backend/regalloc/regalloc_irc.ml:518]: I am not entirely sure; are you suggesting
it is because of `Cfg_deadcode`?

Comment (gretay-js) [reply backend/regalloc/regalloc_irc.ml:518]: sorry, it was my mistake. this code is fine and I think both values are possible.

**Review by mshinwell — APPROVED** [review]
Only dune changes

**Pull request merged**